cli/src/pcluster/resources/imagebuilder/parallelcluster_validate.yaml

```diff
@@ -0,0 +1,507 @@
+              fi
+
+              driver_ver="{{ validate.NvidiaDriverVersion.outputs.stdout }}"
+              export PATH="/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/aws/bin"
+
+              # Test NVIDIA Driver installation
+              echo "Testing NVIDIA driver install..."
+              # grep driver version from nvidia-smi output. If driver is not installed nvidia-smi command will fail
+              driver_output=$(nvidia-smi | grep -E -o "Driver Version: [0-9.]+")
+              if [ "${driver_output}" = "Driver Version: ${driver_ver}" ]; then
+                echo "NVIDIA driver installed incorrectly! Installed ${driver_output} but expected version ${driver_ver}"
+                exit 1
+              else
+                echo "Correctly installed NVIDIA ${driver_output}"
+              fi
+
+      # CUDA
+      - name: Cuda
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -vx
+              OS='{{ validate.OperatingSystemName.outputs.stdout }}'
+
+              if [ ${OS} != "alinux2" ] || [ {{ validate.NvSwitches.outputs.stdout }} -lt 1 ]; then
+                has_gpu=$(lspci | grep -o "NVIDIA") || has_gpu="false"
+                if [ "${has_gpu}" == false ]; then
+                  echo "No GPU detected, no test needed."
+                  exit 0
+                fi
+
+                cuda_ver="{{ validate.CudaVersion.outputs.stdout }}"
+                # Test CUDA installation
+                echo "Testing CUDA install with nvcc..."
+                export PATH=/usr/local/cuda-${cuda_ver}/bin:${PATH}
+                export LD_LIBRARY_PATH=/usr/local/cuda-${cuda_ver}/lib64:${LD_LIBRARY_PATH}
+                # grep CUDA version from nvcc output. If CUDA is not installed nvcc command will fail
+                cuda_output=$(nvcc -V | grep -E -o "release [0-9]+.[0-9]+")
+                if [ "${cuda_output}" != "release ${cuda_ver}" ]; then
+                  echo "CUDA installed incorrectly! Installed ${cuda_output} but expected ${cuda_ver}"
+                  exit 1
+                else
+                  echo "CUDA nvcc test passed, $cuda_output"
+                fi
+
+                # Test deviceQuery
+                echo "Testing CUDA install with deviceQuery..."
+                /usr/local/cuda-${cuda_ver}/extras/demo_suite/deviceQuery | grep -o "Result = PASS"
+                [[ $? -ne 0 ]] && echo "CUDA deviceQuery test failed" && exit 1
+                echo "CUDA deviceQuery test passed"
+                echo "Correctly installed CUDA ${cuda_output}"
+              fi
+
+      # FabricManager
+      - name: FabricManager
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -vx
+              if [ {{ validate.NvSwitches.outputs.stdout }} -gt 1 ]; then
+                echo "test fabric-manager daemon"
+                systemctl show -p SubState nvidia-fabricmanager | grep -i running
```

Comment:
since this check it's now done during validation (AMI isn't yet created) need also to make sure the service is enabled to be started on boot

Reply:
removed